evaluate/task/task-transpile.go

```diff
@@ -0,0 +1,178 @@
+		}
+		modelAssessments.Add(assessments)
+		modelAssessments.Award(metrics.AssessmentKeyResponseNoError)
+
+		coverage, ps, err := targetLanguage.Execute(taskLogger.Logger, packagePath)
+		problems = append(problems, ps...)
+		if err != nil {
+			problems = append(problems, pkgerrors.WithMessage(err, sourceFilePath))
+
+			continue
+		}
+		taskLogger.Printf("Executes tests with %d coverage objects", coverage)
+		modelAssessments.Award(metrics.AssessmentKeyFilesExecuted)
+		modelAssessments.AwardPoints(metrics.AssessmentKeyCoverage, coverage)
+	}
+
+	repositoryAssessment = map[evaltask.Identifier]metrics.Assessments{
+		IdentifierTranspile: modelAssessments,
+	}
+
+	return repositoryAssessment, problems, nil
+}
+
+// unpackTranspilerPackage checks if the testdata repository for the transpilation task is well-formed and returns the path to the implementation file and also the path to the file that holds the stub.
+func (t *TaskTranspile) unpackTranspilerPackage(ctx evaltask.Context, fileLogger *log.Logger, targetLanguage language.Language, packagePath string) (sourceFilePath string, stubFilePath string, err error) {
+	// Check if the package path has a directory called "implementation".
+	var implementationPath string
+	packageFiles, err := os.ReadDir(packagePath)
+	if err != nil {
+		return "", "", pkgerrors.WithStack(err)
+	}
+	for _, file := range packageFiles {
+		if file.IsDir() && file.Name() == "implementation" {
+			implementationPath = filepath.Join(packagePath, file.Name())
+		}
+	}
+	if implementationPath == "" {
+		return "", "", pkgerrors.Errorf("package %q in repository %q must have an \"implementation\" directory with a %s source file to transpile", packagePath, ctx.Repository.Name(), ctx.Language.Name())
+	}
+
+	// Check if the "implementation" directory contains only one source file in the language being transpiled from.
+	implementationFiles, err := os.ReadDir(implementationPath)
+	if err != nil {
+		return "", "", pkgerrors.WithStack(err)
+	} else if len(implementationFiles) != 1 {
+		return "", "", pkgerrors.Errorf("package %q in repository %q must have an \"implementation\" directory with just one %s source file to transpile", packagePath, ctx.Repository.Name(), ctx.Language.Name())
+	}
+	sourceFilePath = filepath.Join(implementationPath, implementationFiles[0].Name())
+	if filepath.Ext(sourceFilePath) != ctx.Language.DefaultFileExtension() {
+		return "", "", pkgerrors.Errorf("package %q in repository %q must have an \"implementation\" directory with a %s source file", packagePath, ctx.Repository.Name(), ctx.Language.Name())
+	} else if strings.HasPrefix(sourceFilePath, ctx.Language.DefaultTestFileSuffix()) {
+		return "", "", pkgerrors.Errorf("package %q in repository %q must have an \"implementation\" directory with only a %s source file, but found a test file %q", packagePath, ctx.Repository.Name(), ctx.Language.Name(), sourceFilePath)
+	}
```

Comment:
Instead of doing all the intermediate checks you can just do `langauge.Files(filepath.Join(packagePath, "implementation"))` and make sure it does not error + returns one implementation file.

Also watch out, your logic for checking if its a test file uses `HasPrefix` :see_no_evil: